packages/xstate-vue/test/UseActorDeferred.vue

```diff
@@ -0,0 +1,28 @@
+<template>
+  <div data-testid="actor-state">
+    {{ actorState.value }}
+  </div>
+  <button @click="actorSend({ type: 'FINISH' })"></button>
+</template>
+
+<script lang="ts">
+import { defineComponent, onMounted } from 'vue';
+import { useActor } from '../src';
+
+export default defineComponent({
+  props: {
+    actor: {
+      type: Object
+    }
+  },
+  setup(props) {
+    const { state: actorState, send: actorSend } = useActor(props.actor as any);
```

Comment:
why do have to use `as any` here? how one could avoid it in the real life project?

Reply:
Tried to improve it with:

``` js
 const { state: actorState, send: actorSend } = useActor(
      props.actor as ActorRefFrom<any>
);
```
If we had the actual machine on the component we could also put like on the React test:
``` js
const { state: actorState, send: actorSend } = useActor(
      props.actor as ActorRefFrom<typeof childMachine>
 );
```

Reply:
Well, the ideal situation would be if could just have `props` typed correctly to avoid `as` assertions altogether (as they are not 100% type-safe). Is there a way to do this?

Reply:
Sorry! I'm very new to Typescript.. I tried with this:
``` vue
<script lang="ts">
import { defineComponent, onMounted, PropType } from 'vue';
import { useActor } from '../src';
import { ActorRefFrom } from 'xstate';

export default defineComponent({
  props: {
    actor: {
      type: Object as PropType<ActorRefFrom<any>>
    }
  },
  setup(props) {
    const { state: actorState, send: actorSend } = useActor(props.actor);

    onMounted(() => {
      actorSend({ type: 'FINISH' });
    });

    return { actorState, actorSend };
  }
});
```

But I was getting an error on props.actor:
```
No overload matches this call.
  Overload 1 of 2, '(actorRef: ActorRef<any, any> | Ref<ActorRef<any, any>>, getSnapshot?: ((actor: ActorRef<any, any>) => any) | undefined): { state: Ref<any>; send: Sender<...>; }', gave the following error.
    Argument of type '(SpawnedActorRef<EventObject, State<unknown, EventObject, any, Typestate<unknown>>> & { state: State<unknown, EventObject, any, Typestate<...>>; }) | undefined' is not assignable to parameter of type 'ActorRef<any, any> | Ref<ActorRef<any, any>>'.
      Type 'undefined' is not assignable to type 'ActorRef<any, any> | Ref<ActorRef<any, any>>'.
  Overload 2 of 2, '(actorRef: ActorRef<EventObject, State<unknown, EventObject, any, Typestate<unknown>>> | Ref<ActorRef<EventObject, State<unknown, EventObject, any, Typestate<...>>>>, getSnapshot?: ((actor: ActorRef<...>) => State<...>) | undefined): { ...; }', gave the following error.
    Argument of type '(SpawnedActorRef<EventObject, State<unknown, EventObject, any, Typestate<unknown>>> & { state: State<unknown, EventObject, any, Typestate<...>>; }) | undefined' is not assignable to parameter of type 'ActorRef<EventObject, State<unknown, EventObject, any, Typestate<unknown>>> | Ref<ActorRef<EventObject, State<unknown, EventObject, any, Typestate<...>>>>'.
      Type 'undefined' is not assignable to type 'ActorRef<EventObject, State<unknown, EventObject, any, Typestate<unknown>>> | Ref<ActorRef<EventObject, State<unknown, EventObject, any, Typestate<...>>>>'.Vetur(2769)
```

Do you have any suggestion?

Reply:
I've pushed out some commits into your branch - so please pull them before you continue working on this branch.

I've removed this as assertion from another file (since this one was removed) and it seems to work. Please recheck if it's OK on your machine now (if not maybe try in a fresh copy of the repo? or rebuild the monorepo?).

Reply:
Thanks!! It's interesting...
If done a fresh copy of the repo but had the same errors. But have found that:
- if I open the repo on VS Code from the xstate-vue package folder, where there's a tsconfig.json, Vetur gives this Typescript errors.
- If i open the folder from the root xstate folder, Vetur gives an initial startup error that can't find a tsconfig, but then I don't have this Typescript errors... So I'll continue working this way :)